utils/build_swift/argparse/types.py

```diff
@@ -32,6 +34,44 @@
 
 # -----------------------------------------------------------------------------
 
+class CompilerVersion(object):
```

Comment:
Do we already have anything like this?

Reply:
We do have an equivalent (albeit less type-safe) class in `utils/swift_build_support/swift_build_support/arguments.py`, however one of my main goals with the `build_swift` module is to completely remove reliance on that package since it's quite messy. Rather I'm opting here to implement a well-tested version analogous to the older one.

Reply:
We should add a comment to the old one indicating that it's deprecated then and link to the new one.

Reply:
Sure thing!